### ISSUE TITLE
Additional step to stop recording

### ISSUE THREAD

**Issue by downtownallday** [y20k/trackbook#92]
I love the program, thank you!

This is an enhancement request to make it harder to stop recording.

I use it hiking to keep track of my accomplishments. All works fairly well, but I find it too easy to mistakenly press the stop recording button especially in cold weather, where it gets trickier to manipulate the phone with frozen fingers.

I accidentally pressed the stop recording button yesterday and didn't notice until a couple miles down the trail.

It would be great if the stop recording button (circle with red dot in it), when pressed, instead of immediately pausing/stopping recording, opened a menu (just like the one for save/clear/resume), except the menu would have a single option for "pause" or "stop recording".

Thanks

**Comment by y20k** [comment]
@downtownallday 

First of all. Happy new year.

There are three possible ways to stop a recording: 

1. via the app
2. via the notification
3. and (if set up) via Quick Settings. 

All of those stop a recording immediately. I can only show a confirmation dialog in the app itself. Meaning that I can only secure one possible way of triggering Stop Recording. In the app a color change in the recorded waypoints and the recording button already indicates the changed state. And you can correct your error easily by hitting Resume.

So basically I do not see the necessity for a change here, especially since users might be annoyed when they see a dialog.

But what about showing a [Toast Message](https://developer.android.com/guide/topics/ui/notifiers/toasts) "Recording Started" "Recording Stopped"?

**Comment by downtownallday** [comment]
Thanks! I was referring to the red stop recording button in the app itself. It's just too easy to press it. It happened again last week while my phone was in my pocket or backpack. Somehow the phone came on and the stop recording button got pressed "by itself". A toast is fine, but won't help solve this problem I'm having.

I'll try to be more careful when using the app (zoom, unzoom, pan, etc are very close to the stop recording button), and also remember to activate a different app before putting the phone back into my pocket.

**Comment by Hob111** [comment]
Surely if you shut off the screen it's impossible to stop the app while it's in your pocket?

**Comment by y20k** [comment]
> It happened again last week while my phone was in my pocket or backpack. 

Then recording was probably stopped via 2) Notification or 3) Quick Setting.

Or the app crashed due to some reason.

**Comment by Hob111** [comment]
Don't you shut off the screen and have a unlock code of some sort? 

On 11 January 2021 20:02:08 GMT, y20k <notifications@github.com> wrote:
> It happened again last week while my phone was in my pocket or
backpack. 

Then recording was probably stopped via 2) Notification or 3) Quick
Setting.

Or the app crashed due to some reason.

**Comment by downtownallday** [comment]
Yes, I shut the screen off and have lock pattern. However, I've configured the phone so the lock doesn't activate until 2 minutes after pressing the screen off button. Yes I could change that, too. Ideally though, I would prefer a much longer delay for the lock to activate while hiking. Entering the pattern when hands are frigid cold is a real pain(!). 

App did not crash as far as I could tell.

**Comment by Hob111** [comment]
I'm a little puzzled why you need such a long delay before the screen is locked. I start the app, shut off the screen and put the phone in my pocket. I can think of very few circumstances where I might want to access the phone again in the next two minutes. Surely you'd be better having the screen locked for the 99 times you don't return to the phone within two minutes rather than having it unlocked for the one time you do. Have I missed something? 🤔

**Comment by downtownallday** [comment]
Switching to OpenTracks
